notus/scanner/loader/gpg_sha_verifier.py

```diff
@@ -0,0 +1,75 @@
+import hashlib
+import os
+from pathlib import Path
+from typing import Callable, Dict, Optional
+
+from gnupg import GPG
+
+
+class GPGException(Exception):
```

Comment:
Then it would not be useable in ospd or other projects

Reply:
And it is already catched in the json loader.

Reply:
It is used in the daemon.py which as far as I can tell doesn't do error handling so it would just crash in that case and it shouldn't affect runtime behavior when it is not a notus error.

Reply:
Alternative is to not verify that eagerly and let the actual gnupg module throw an exception; the reasoning for checking eagerly is more optimization based since we use that path to fulfill other requirements but would fail in any case; so that the reason for that check is a weak.

Reply:
That would mean to also change the return value to Opional[Dict[str, str]] to return None when verification fail and let the daemon decide what to do.

Reply:
Implemented alternative and add a notus specific error for daemon.py